File: src/auth/auth_ticket/_ui/clear/event.ts
import { RepositoryError } from "../../../../z_details/_ui/repository/data"
import { ClearAuthTicketError } from "./data"

export type ClearAuthTicketEvent =
    | Readonly<{ type: "failed-to-logout"; err: RepositoryError }> // TODO これは repository-error にするべき
    | Readonly<{ type: "failed-to-clear"; err: ClearAuthTicketError }>
    | Readonly<{ type: "succeed-to-logout" }>

File: src/auth/ticket/remote/discard/event.rs
use crate::auth::ticket::remote::validate::event::ValidateAuthTokenEvent;

use crate::z_details::_common::repository::data::RepositoryError;

pub enum DiscardAuthTicketEvent {
    Success,
    Validate(ValidateAuthTokenEvent),
    RepositoryError(RepositoryError),
}

const SUCCESS: &'static str = "discard success";
const ERROR: &'static str = "discard error";

impl std::fmt::Display for DiscardAuthTicketEvent {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            Self::Success => write!(f, "{}", SUCCESS),
            Self::Validate(event) => event.fmt(f),
            Self::RepositoryError(err) => write!(f, "{}: {}", ERROR, err),
        }
    }
}
